os/arch/arm/src/common/up_createstack.c

```diff
@@ -263,9 +265,11 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 #endif
 		board_led_on(LED_STACKCREATED);
 #ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
-		/* The smallest size that can be programmed for an MPU region is 32 bytes */
-		mpu_get_register_config_value(&tcb->stack_mpu_regs[0], MPU_REG_NUM_STK,
-			(uint32_t)tcb->stack_alloc_ptr, CONFIG_MPU_STACK_GUARD_SIZE, true, false);
+		if (sched_self()->uheap) {
```

Comment:
You call `sched_self()->uheap` several times. It looks redundant. How about checking it once and using?

Reply:
And could you leave the comment to say why sched_self is used instead of tcb given?
This function creates the stack of new thread so that it should be checked with new thread's heap. But currently it checks parent's heap value.
1. new thread does not have the uheap value yet. (It's added in the task_schedsetup function)
2. parent and child threads must have the same uheap value.

Reply:
modified, thanks